Lambdas/GHHooks/EventHandler/Handlers/ReleaseHandler.swift

```diff
@@ -0,0 +1,414 @@
+        newVersion: String
+    ) async throws -> String {
+        let codeOwners = try await getCodeOwners()
+        let contributors = try await getExistingContributorIDs()
+        let isNewContributor = isNewContributor(
+            codeOwners: codeOwners,
+            existingContributors: contributors
+        )
+        let reviewers = try await getReviewersToCredit(codeOwners: codeOwners)
+        let isCodeOwner = codeOwners.contains(pr.user.login) ||
+        (pr.user.name.map { codeOwners.contains($0) } ?? false)
+
+        return """
+        \(makeMergerMarkdown(mergedBy: mergedBy))
+        \(makePRMarkdown(isCodeOwner: isCodeOwner))
+        \(makeContributorMarkdown(isNewContributor: isNewContributor))
+        \(makeReviewersMarkdown(reviewers: reviewers))
+        \(makeChangeLogMarkdown(previousVersion: previousVersion, newVersion: newVersion))
+        """
+    }
+
+    func makeMergerMarkdown(mergedBy: NullableUser) -> String {
+        """
+        ###### _This patch was released by @\(mergedBy.name ?? mergedBy.login)._
+
+        """
+    }
+
+    func makePRMarkdown(isCodeOwner: Bool) -> String {
+        if isCodeOwner { return "" }
+        let body = pr.body.map {
+            "> " + $0.formatMarkdown(
+                maxLength: 512,
+                trailingParagraphMinLength: 128
+            ).replacingOccurrences(
+                of: "\n",
+                with: "\n> "
+            )
+        } ?? ""
+        return """
+        ## What's Changed
+        \(pr.title) by @\(pr.user.name ?? pr.user.login) in #\(number)
+
+        \(body)
+
+        """
+    }
+
+    func makeContributorMarkdown(isNewContributor: Bool) -> String {
+        guard isNewContributor else { return "" }
+        return """
+        ## New Contributor
+        - @\(pr.user.name ?? pr.user.login) made their first contribution 🎉
+        
+        """
+    }
+
+    func makeReviewersMarkdown(reviewers: [User]) -> String {
+        if reviewers.isEmpty { return "" }
+        let reviewersText = reviewers.map { user in
+            "- @\(user.name ?? user.login)"
+        }.joined(separator: "\n")
+        return """
+        ## Reviewers
+        Thanks to the reviewers:
+        \(reviewersText)
+
+        """
+    }
+
+    func makeChangeLogMarkdown(previousVersion: String, newVersion: String) -> String {
+        let fullName = repo.full_name.addingPercentEncoding(
+            withAllowedCharacters: .urlPathAllowed
+        ) ?? repo.full_name
+        let url = "https://github.com/\(fullName)/compare/\(previousVersion)...\(newVersion)"
+        return "**Full Changelog**: \(url)"
+    }
+
+    func getReviewersToCredit(codeOwners: Set<String>) async throws -> [User] {
+        let reviewComments = try await getReviewComments()
+        let reviewers = reviewComments.map(\.user).filter { user in
+            !codeOwners.contains(user.login) &&
+            !(user.name.map { codeOwners.contains($0) } ?? false)
+        }
```

Comment:
i don't know what was i thinking when i decided to do `!&!` instead of just `||`.... i think i just wanted to not use `||` but i don't think that was the right call.